File: app/src/main/java/com/drawable/Text.java
package com.drawable;

import android.graphics.Canvas;
import android.graphics.Paint;

import com.util.Color;

import java.math.BigDecimal;

@SuppressWarnings({"unused", "WeakerAccess"})
public class Text extends Drawable {
    private String text;
    private double textSize;

    public Text(String text, double x, double y, double textSize, int color) {
        super(x, y, color);
        this.text = text;
        this.textSize = textSize;
    }

    public Text(String text, double x, double y, double textSize, Color color) {
        this(text, x, y, textSize, color.getInt());
    }

    public Text(Text text) {
        super(text);
        this.text = text.getText();
        this.textSize = text.getTextSize();
    }

    public String getText() {
        return text;
    }

    public double getTextSize() {
        return textSize;
    }

    public void setText(String text) {
        this.text = text;
    }

    public void setTextSize(double textSize) {
        this.textSize = textSize;
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof Text)) return false;
        Text other = (Text) o;
        BigDecimal first = new BigDecimal(Double.toString(this.textSize));
        BigDecimal second = new BigDecimal(Double.toString(other.textSize));
        return this.text.equals(other.text) && first.compareTo(second) == 0 && this.getColor() == other.getColor();
    }

    @Override
    public void draw(Canvas canvas, Paint paint) {
        paint.setStyle(Paint.Style.FILL);
        paint.setTextSize((float) textSize);
        paint.setColor(getColor());
        canvas.drawText(text, (float) getX(), (float) getY(), paint);
        animate();
    }

    @Override
    public String toString() {
        String s = super.toString();
        return "Text: \n \tMessage: " + text + "\n \tText size: " + textSize + "\n" +
                s.substring(this.getClass().getSimpleName().length() + 3, s.indexOf("X velocity: "));
    }
}

File: app/src/main/java/com/drawable/Text.kt
package com.drawable

import android.graphics.Canvas
import android.graphics.Paint

import com.util.Color

import java.math.BigDecimal

@Suppress("unused", "NAME_SHADOWING", "EqualsOrHashCode")
class Text : Drawable {
    var text: String = ""
    var textSize: Double = 0.toDouble()

    constructor(text: String, x: Double, y: Double, textSize: Double, color: Int) : super(x, y, color) {
        this.text = text
        this.textSize = textSize
    }

    constructor(text: String, x: Double, y: Double, textSize: Double, color: Color) : this(text, x, y, textSize, color.int)

    constructor(text: Text) : super(text) {
        this.text = text.text
        this.textSize = text.textSize
    }

    override fun equals(other: Any?): Boolean {
        if (other !is Text) return false
        val other = other as Text?
        val first = BigDecimal(java.lang.Double.toString(this.textSize))
        val second = BigDecimal(java.lang.Double.toString(other!!.textSize))
        return this.text == other.text && first.compareTo(second) == 0 && this.color == other.color
    }

    override fun draw(canvas: Canvas, paint: Paint) {
        paint.style = Paint.Style.FILL
        paint.textSize = textSize.toFloat()
        paint.color = color
        canvas.drawText(text, x.toFloat(), y.toFloat(), paint)
        animate()
    }

    override fun toString(): String {
        val s = super.toString()
        return "Text: \n \tMessage: " + text + "\n \tText size: " + textSize + "\n" +
                s.substring(this.javaClass.simpleName.length + 3, s.indexOf("X velocity: "))
    }
}